awx/ui_next/src/screens/Inventory/shared/InventoryGroupsDeleteModal.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import AlertModal from '@components/AlertModal';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { Button, Radio } from '@patternfly/react-core';
+import styled from 'styled-components';
+
+const ListItem = styled.div`
+  padding: 24px 1px;
+
+  dl {
+    display: flex;
+    font-weight: 600;
+  }
+  dt {
+    color: var(--pf-global--danger-color--100);
+    margin-right: 10px;
+  }
+  .pf-c-radio {
+    margin-top: 10px;
+  }
+`;
+
+const ContentWrapper = styled.div`
+  ${ListItem} + ${ListItem} {
+    border-top-width: 1px;
+    border-top-style: solid;
+    border-top-color: #d7d7d7;
+  }
+  ${ListItem}:last-child {
+    padding-bottom: 0;
+  }
+ `;
+
+const InventoryGroupsDeleteModal = ({
+  onClose,
+  onDelete,
+  isModalOpen,
+  groups,
+  i18n,
+}) => {
+  const [deleteList, setDeleteList] = useState([]);
+
+  useEffect(() => {
+    const groupIds = groups.reduce((obj, group) => {
+      if (group.total_groups > 0 || group.total_hosts > 0) {
+        return { ...obj, [group.id]: null };
+      }
+      return { ...obj, [group.id]: 'delete' };
+    }, {});
+
+    setDeleteList(groupIds);
+  }, [groups]);
+
+  const handleChange = (groupId, radioOption) => {
+    setDeleteList({ ...deleteList, [groupId]: radioOption });
+  };
+
+  const content = groups
+    .map(group => {
+      if (group.total_groups > 0 || group.total_hosts > 0) {
+        return (
+          <ListItem key={group.id}>
+            <dl>
+              <dt>{group.name}</dt>
+              <dd>
+                {i18n._(
+                  t`(${group.total_groups} Groups and ${group.total_hosts} Hosts)`
```

Comment:
I think this is referencing fields that won't exist soon (`group.total_hosts`, `group.total_groups`)

cc @ryanpetrello @marshmalien

Reply:
Updated in 0d3ec055d9ce888e14b5d6e15ac4ad29f1cbf74d

We decided to remove the (N Groups and M Hosts) on the label and force the user to either Delete or Promote the children of selected groups. See gif in summary.